Userland/Services/AudioServer/Mixer.cpp

```diff
@@ -1,29 +1,40 @@
 /*
  * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021, kleines Filmröllchen <malu.bertsch@gmail.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include "Mixer.h"
 #include <AK/Array.h>
 #include <AK/MemoryStream.h>
 #include <AK/NumericLimits.h>
 #include <AudioServer/ClientConnection.h>
 #include <AudioServer/Mixer.h>
+#include <LibCore/ConfigFile.h>
+#include <LibCore/Timer.h>
 #include <pthread.h>
 #include <strings.h>
 
 namespace AudioServer {
 
 u8 Mixer::m_zero_filled_buffer[4096];
 
-Mixer::Mixer()
+Mixer::Mixer(NonnullRefPtr<Core::ConfigFile> config)
     : m_device(Core::File::construct("/dev/audio", this))
     , m_sound_thread(Threading::Thread::construct(
           [this] {
               mix();
               return 0;
           },
           "AudioServer[mixer]"))
+    , m_config(move(config))
+    , m_config_write_timer(Core::Timer::create_repeating(
```

Comment:
This will cause an unnecessary AudioServer wake-up every 10 seconds, even when there's no sound playing or any config changes to write out.
Let's use a single-shot timer instead, and start it whenever something configurable is changed.